app/controllers/seasons_controller.rb

```diff
@@ -0,0 +1,56 @@
+class SeasonsController < ApplicationController
+  def index
+    @film = Film.find(params[:id])
+    @seasons = @film.seasons.all
+  end
+
+  def show
+    @season = Season.find(params[:id])
+    @film = Film.find(@season.film_id)
+    @episodes = @season.episode.all
+  end
+
+  def new
+    @film = Film.find(params[:id])
+    @season = @film.seasons.new
+  end
+
+  def create
+    @film = Film.find(params[:id])
+    @season = @film.seasons.new(season_params)
+
+    if @season.save
+      redirect_to @season
+    else
+      render :new
+    end
+  end
```

Comment:
```ruby
before_action :find_film, only: %i[index new create]
private

def find_film
    @film = Film.find(params[:film_id])
end

```

Reply:
I rewrote the controller logic from nested to resource.